Sources/GitBuddyCore/Models/Changelog.swift

```diff
@@ -9,12 +9,40 @@
 import Foundation
 import OctoKit
 
-struct Changelog: CustomStringConvertible {
-    typealias PullRequestID = Int
-    typealias IssueID = Int
-    let description: String
+typealias PullRequestID = Int
+typealias IssueID = Int
 
+protocol Changelog: CustomStringConvertible {
     /// The pull requests ID and related issues IDs that are merged with the related release of this changelog.
+    var itemIdentifiers: [PullRequestID: [IssueID]] { get }
+}
+
+struct SectionedChangelog: Changelog {
+    let description: String
+
+    let itemIdentifiers: [PullRequestID: [IssueID]]
+    
+    init(issues: [Issue], pullRequests: [PullRequest]) {
+        description =
+            """
+            **Closed issues:**
+
+            \(ChangelogBuilder(items: issues.map { ChangelogItem(input: $0, closedBy: $0) }).build())
+
+            **Merged pull requests:**
+
+            \(ChangelogBuilder(items: pullRequests.map { ChangelogItem(input: $0, closedBy: $0) }).build())
+            """
+
+        self.itemIdentifiers = pullRequests.reduce(into: [:]) { (result, item) in
```

Comment:
```suggestion
        itemIdentifiers = pullRequests.reduce(into: [:]) { (result, item) in
```

Reply:
Would it make sense to set up [SwiftFormat](https://github.com/nicklockwood/SwiftFormat) with a config that would apply a rule like this automatically (in a separate PR)?

Reply:
I would love to see that, yes! We're not yet familiar with SwiftFormat at WeTransfer, so. itwould be great if you could introduce us to that with a PR solving this issue :)